From e9607769d56373cf29c4df040f605191603beb50 Mon Sep 17 00:00:00 2001
From: Nenad Merdanovic <nmerdan@haproxy.com>
Date: Sun, 12 Mar 2017 22:01:35 +0100
Subject: [PATCH 5/7] BUG/MEDIUM: cli: Prevent double free in CLI ACL lookup

The memory is released by cli_release_mlook, which also properly sets the
pointer to NULL. This was introduced with a big code reorganization
involving moving to the new keyword registration form in commit ad8be61c7.

This fix needs to be backported to 1.7.

Signed-off-by: Nenad Merdanovic <nmerdan@haproxy.com>
(cherry picked from commit 24f45d8e34797ed9c16ac3fa6d89f3eed435e706)
---
 src/map.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/map.c b/src/map.c
index b6fce4d..90deb08 100644
--- a/src/map.c
+++ b/src/map.c
@@ -524,7 +524,6 @@ static int cli_io_handler_map_lookup(struct appctx *appctx)
 
 	default:
 		appctx->st2 = STAT_ST_FIN;
-		free(appctx->ctx.map.chunk.str);
 		return 1;
 	}
 }
-- 
2.10.2

